src/Streamly/Internal/Data/Fold.hs

```diff
@@ -556,17 +556,17 @@ stdDev = sqrt variance
 --
 -- @since 0.7.0
 {-# INLINABLE rollingHashWithSalt #-}
-rollingHashWithSalt :: (Monad m, Enum a) => Int -> Fold m a Int
+rollingHashWithSalt :: (Monad m, Enum a) => Integer -> Fold m a Integer
 rollingHashWithSalt salt = Fold step initial extract
     where
-    k = 2891336453
+    k = 2891336453 :: Integer
     initial = return salt
-    step cksum a = return $ cksum * k + fromEnum a
+    step cksum a = return $ cksum * k + fromIntegral (fromEnum a)
     extract = return
 
 -- | A default salt used in the implementation of 'rollingHash'.
 {-# INLINE defaultSalt #-}
-defaultSalt :: Int
+defaultSalt :: Integer
 #if WORD_SIZE_IN_BITS == 64
```

Comment:
I guess this can be removed if we use `Int64` type here. @adithyaov

Reply:
`WORD_SIZE_IN_BITS` is undefined on my 64bit machine... what's going on here.

Reply:
And it doesn't work:

```
src/Streamly/Internal/Data/Fold.hs:571:15: warning: [-Woverflowed-literals]
    Literal 15868100553162883236 is out of the Int64 range -9223372036854775808..9223372036854775807
    |
571 | defaultSalt = 0xdc36d1615b7400a4
```

Reply:
I am not sure where that conditional came from, it probably remained undefined all the time.

Reply:
Yes, and the default salt doesn't fit into Int64.

Reply:
0xdc36d1615b7400a4 is not correct, it should be a negative int value of (-2578643520546668380). If we write that in Hex it turns out to be 0xdc36d1615b7400a4 because of the sign bit. So the correct code here will be:

```
defaultSalt = -2578643520546668380
```

Reply:
Done